resolwe_runtime_utils.py

```diff
@@ -275,14 +301,224 @@ def checkrc(rc, *args):
+        temp_dir = 'temp_{}'.format(extracted_name)
+
+        cmd = '7z x -y -o{} {}'.format(shlex.quote(temp_dir), shlex.quote(src))
+        subprocess.check_call(cmd, shell=True)
+
+        paths = os.listdir(temp_dir)
+        if len(paths) == 1 and os.path.isfile(os.path.join(temp_dir, paths[0])):
+            # Single file in archive.
+            temp_file = os.path.join(temp_dir, paths[0])
+
+            if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
+                with open(temp_file, 'rb') as f_in, gzip.open(
+                    extracted_name + '.gz', 'wb'
+                ) as f_out:
+                    shutil.copyfileobj(f_in, f_out, CHUNK_SIZE)
+
+            if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+                shutil.move(temp_file, './{}'.format(extracted_name))
+
+                if extracted_name.endswith('.tar'):
+                    with tarfile.open(extracted_name) as tar:
+                        tar.extractall()
+
+                    os.remove(extracted_name)
+                    extracted_name, _ = os.path.splitext(extracted_name)
+        else:
+            # Directory or several files in archive.
+            if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
+                with tarfile.open(extracted_name + '.tar.gz', 'w:gz') as tar:
+                    for fname in glob.glob(os.path.join(temp_dir, '*')):
+                        tar.add(fname, os.path.basename(fname))
+
+            if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+                for path in os.listdir(temp_dir):
+                    shutil.move(os.path.join(temp_dir, path), './{}'.format(path))
+
+        shutil.rmtree(temp_dir)
+        return (
+            extracted_name + '.gz'
+            if imported_format == ImportedFormat.COMPRESSED
+            else extracted_name
+        )
+
+    def importUncompressed():
+        """Import uncompressed file."""
+        if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
+            with open(src, 'rb') as f_in, gzip.open(file_name + '.gz', 'wb') as f_out:
+                shutil.copyfileobj(f_in, f_out, CHUNK_SIZE)
+
+        if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+            try:
+                shutil.copyfile(src, file_name)
+            except shutil.SameFileError:
+                pass  # Skip copy of downloaded files
+
+        return (
+            file_name + '.gz'
+            if imported_format == ImportedFormat.COMPRESSED
+            else file_name
+        )
+
+    # Large file download from Google Drive requires cookie and token.
+    response = None
+    if re.match(
+        r'^https://drive.google.com/[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]$',
+        src,
+    ):
+        session = requests.Session()
+        response = session.get(src, stream=True)
```

Comment:
Uncaught exception.

Reply:
I've decided to reverse the logic and change `print(error(...))` with raise exception. I will update Python processes to catch exceptions and automatically save process errors.